numpy/core/tests/test_multiarray.py

```diff
@@ -4541,6 +4541,16 @@ def test_file_position_after_tofile(self):
             f.close()
             assert_equal(pos, 10, err_msg=err_msg)
 
+    def test_allocation_error_fromfile(self):
+        # gh-12300
+        with open(self.filename, 'wb') as f:
+            f.write(b'\xfe')
+
+        with open(self.filename, 'rb') as f:
+            assert_raises(ValueError, np.fromfile, f, np.unicode)
```

Comment:
This could do with a comment. Is there anything special about the value `\xfe`, or is the important thing that it's less than the 32 bits expected by a unicode character?

Reply:
Hrmm on second thoughts, this test isn't great.

It's failing for the wrong reason: I thought it was because of the reason you were saying. However instead it looks like it's because, `np.unicode` has a size of 0 and we are getting a ValueError as `PyArray_FromFile` tries to allocate an array of -1 unicode objects.

This looks like a separate issue in and of itself, and so we probably shouldn't rely on it for testing this case.

It's not obvious how to test this though, I can't quite pin a case which will get `PyArray_FromFile` to raise something consistently.

Reply:
ok, scratch that. Turns out that `PyArray_FromFile` will consistently raise a `ValueError` for a good reason when asked to read into an object array.

I've changed the test to do that.